Code/Source/RobotImporter/URDF/RobotImporter.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace ROS2
+{
+    class RobotImporterWidget;
+
+    class RobotImporter
+    {
+    public:
+        explicit RobotImporter(RobotImporterWidget& robotImporterWidget);
```

Comment:
RobotImporter could be designed so that it also work without the widget - for example, within the Asset Processor. 
This file could include RobotImporterConfiguration which would contain all that is necessary, except for the GUI element for logging. Consider using optional std::function(s) for logging, which could be set accordingly when constructing with the Widget (by default they would only log with to console with AZ_TracePrintf or similar)

Reply:
Ok looking further I see that there is still URDFPrefabMaker which could be used independently from the Widget. Consider points in the comment above but feel free to disregard.